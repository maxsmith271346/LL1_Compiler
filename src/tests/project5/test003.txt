main

int x;

function foo(int n) : void 
{
    return n;
};

function foo(int n, float d) : int 
{
    call printInt(10+n);
    call printFloat(10+d);
};

function bar(int n) : void
{
    int x;
    call printInt(n*10);
    call println();
    let x = call foo(n);
    call printInt(x+0);
    call printInt();
};

{
    call bar(10.0);
}.