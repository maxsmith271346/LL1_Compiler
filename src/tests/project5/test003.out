Error type-checking file.
TypeError(7,5)[Function foo returns int instead of void.]
TypeError(13,23)[Cannot add int to float.]
TypeError(13,5)[Call with args TypeList(ErrorType(Cannot add int to float.)) matches no function signature.]
TypeError(21,5)[Cannot assign void to AddressOf(int).]
TypeError(23,5)[Call with args TypeList() matches no function signature.]
TypeError(27,5)[Call with args TypeList(float) matches no function signature.]
